classes/migration/upgrade/v3_4_0/I11125_UpdateEmailTemplateVariables.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I11125_UpdateEmailTemplateVariables.php
+ *
+ * Copyright (c) 2014-2025 Simon Fraser University
+ * Copyright (c) 2000-2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I11125_UpdateEmailTemplateVariables
+ *
+ * @brief Migration to update Email Template variable names
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Support\Facades\DB;
+
+abstract class I11125_UpdateEmailTemplateVariables extends \PKP\migration\Migration
+{
+    public function up(): void
+    {
+        // Update template variables
+        $this->renameTemplateVariables($this->oldToNewVariablesMap());
+    }
+
+    public function down(): void
+    {
+        $newToOldVariableMap = array_map(function ($variablesMap) {
+            return array_flip($variablesMap);
+        }, $this->oldToNewVariablesMap());
+
+        $this->renameTemplateVariables($newToOldVariableMap);
+    }
+
+    /**
+     * Replaces email template variables in templates' subject and body
+     */
+    protected function renameTemplateVariables(array $oldNewVariablesMap): void
+    {
+        foreach ($oldNewVariablesMap as $emailKey => $variablesMap) {
+            $existingVariables = [];
+            $replacementsVariables = [];
+
+            foreach ($variablesMap as $key => $value) {
+                $existingVariables[] = '/\{\$' . $key . '\}/';
+                $replacementsVariables[] = '{$' . $value . '}';
+            }
+
+            // Default templates
+            $data = DB::table('email_templates_default_data')->where('email_key', $emailKey)->get();
```

Comment:
I'm not sure how many rows this and the next table might have (some installations out there have many journals), but such simple string replacements can be done with SQL, then you won't need to download all the record set and the operation should complete faster.